editor/src/clj/editor/gui.clj

```diff
@@ -2867,5 +2924,14 @@
   (:node-cls (get-registered-node-type-info node-type custom-type)))
 
 ;; SDK api
-(defn register-node-type-info! [type-info]
+(defn register-node-type-info! [{:keys [node-cls] :as type-info}]
+  (when-some [abstract-output-labels (not-empty (g/abstract-output-labels node-cls))]
+    (throw (ex-info (format "Plugin GUI node type %s does not implement required outputs: %s"
+                            (:name @node-cls)
+                            (->> abstract-output-labels
+                                 (sort)
+                                 (map name)
+                                 (str/join ", ")))
+                    {:node-cls node-cls
+                     :abstract-output-labels abstract-output-labels})))
```

Comment:
Added check to ensure any `GuiNode` subclasses registered by extensions implement all abstract outputs so you get the error message early instead of when building / saving.